### PR TITLE
Adding custom component props to FormField component

### PR DESCRIPTION
**Pull request by georgewrmarshall**
- [ ] Needs unit tests

## Explanation

Currently, the `FormField` component doesn't allow for any custom components.

This is a problem because if we want to extend the `FormField` components functionality by using a different type of label or unit or tooltip. It's not possible.

In order to solve this problem, this pull request adds custom component props to allow for the default components to be replaced.

### After

Screen Shot 2022-08-23 at 9 29 47 AM

## Manual Testing Steps

- Go to the latest storybook build of this PR 
- Search `FormField`
- Look at the Custom Components story

## Pre-Merge Checklist

- [x] PR template is filled out
- ~[x] **IF** this PR fixes a bug, a test that _would have_ caught the bug has been added~ N/A
- ~[x] PR is linked to the appropriate GitHub issue~ N/A
- ~[x] PR has been added to the appropriate release Milestone~ N/A

### + If there are functional changes:

- [ ] Manual testing complete & passed
- ~[ ] "Extension QA Board" label has been applied~ N/A

### CODE REVIEW

File: ui/components/ui/form-field/form-field.js
Comment (georgewrmarshall) [inline ui/components/ui/form-field/form-field.js:27]: Using PascalCase prop names here to convey that components should be passed to them. Happy to change to camelCase if it makes more sense to.

File: ui/components/ui/form-field/form-field.stories.js
Comment (georgewrmarshall) [inline ui/components/ui/form-field/form-field.stories.js:116]: @VSaric example of how this component could work with your use case https://github.com/MetaMask/metamask-extension/pull/15522

Comment (VSaric) [reply ui/components/ui/form-field/form-field.stories.js:116]: @georgewrmarshall Wow, this is amazing! Thank you for these updates that could work with my use case! But I have couple of questions. I used this in `form-field.stories.js`:
```
 titleDetail={
          <button style={{
            backgroundColor: clicked
              ? 'var(--color-warning-default)'
              : 'var(--color-background-alternative)',
          }}
          onClick={() => setClicked(!clicked)}
          >
            TitleDetail
          </button>
        }
        titleDetailWrapperProps={{ marginBottom: 0 }}
        customSpendingCapText={<Typography>Only enter a number that you're comfortable with the contract spending now or in the future. You can always increase the spending cap later.</Typography>}
```
I added in `titleDetail` property `button`, not `Typography` like you have here. I added that and `customSpendingCapText` property because of the use case that I must have in #15522. 

My question is, when `titleDetail` property have `button`, why it is clickable everywhere I click? It is maybe a small bug or this scenario is okay? You can see this in the screencast below:


https://user-images.githubusercontent.com/92527393/186387909-aa9e1d50-43cb-4822-b078-d6e74f888fb9.mov

I noticed that all `form-field` component is wrapped around `<label>...</label>` and because of that I think when I have in `titleDetail` property `button`, it is clickable everywhere I click.

Comment (georgewrmarshall) [reply ui/components/ui/form-field/form-field.stories.js:116]: Ah yes you're absolutely right I've removed the wrapping label component and added it to just the `titleText` here https://github.com/MetaMask/metamask-extension/pull/15679/commits/b0dd98f6ec3ba2523ab5968f2548fd407771b532

**Comment by VSaric** [comment]
I saw one other thing, that the `test-unit` is failing. I run unit test in this branch and the tests are failing in `networks-form.test.js`: 

Screenshot 2022-08-25 at 10 39 21

My suggestion to fix this tests is to add `<label></label>` between the `FormField` components in `networks-form.js`. It is look like this (example): 

> 
        <label>
          <FormField
            autoFocus
            error={errors.networkName?.msg || ''}
            onChange={(value) => {
              setIsEditing(true);
              setNetworkName(value);
            }}
            titleText={t('networkName')}
            value={networkName}
            disabled={viewOnly}
          />
        </label>

In the below screenshot I run `npx jest networks-form.test.js` to run only that test to see if all tests pass:

Screenshot 2022-08-25 at 10 49 21

If we add this than the tests pass and pipeline will be ✅ . If you have another solution for fixing this tests please do that. 😄

File: ui/components/ui/form-field/form-field.stories.js
Comment (VSaric) [reply ui/components/ui/form-field/form-field.stories.js:116]: @georgewrmarshall This is excellent! I posted one more comment for failing unit tests. When tests pass, do you want to open this PR and merge it to the develop? I just wondering because if this PR will go on develop, then I must rebased and wait until this PR was merged to finish my use case in #15522.

Comment (georgewrmarshall) [reply ui/components/ui/form-field/form-field.stories.js:116]: Yeah apologies this is a blocker. Will work on this today

File: ui/components/ui/typography/typography.js
Comment (georgewrmarshall) [inline ui/components/ui/typography/typography.js:49]: Allows for the root html element of the `Typography` component to be rendered as a `<label />`

**Review by VSaric — APPROVED** [review]
LGTM!

**Review by darkwing — APPROVED** [review]

**Review by garrettbear — APPROVED** [review]

**Comment by georgewrmarshall** [comment]
Going to merge to unblock and create unit tests in another PR

**Pull request merged**
